File: a1-programming-basics-javascript/06b-nested-loops-exercise/04-train-the-trainers.js
function trainTheTrainers(input) {
  let judges = Number(input[0]);
  let index = 1;
  let name = input[1];
  index++;
  let sum = 0;
  let bigSum = 0;
  let counter = 0;
  let bigCounter = 0;

  while (name !== `Finish`) {
    let score = Number(input[index]);
    sum += score;
    counter++;

    if (counter === judges) {
      bigSum += sum;
      bigCounter += counter;
      sum = sum / judges;
      console.log(`${name} - ${sum.toFixed(2)}.`);
      counter = 0;
      name = input[index + 1];
      sum = 0;
      index += 1;
    }
    index++;
  }

  console.log(
    `Student's final assessment is ${(bigSum / bigCounter).toFixed(2)}.`
  );
}

File: a01-softuni-programming-basics-python/06b-nested-loops-exercise/04_train_the_trainers.py
juri = int(input())
presentation = input()
total_average = 0
count_presentation = 0

while presentation != "Finish":
    average_assessment = sum(float(input()) for _ in range(juri))
    average_assessment /= juri
    print(f"{presentation} - {average_assessment:.2f}.")
    total_average += average_assessment
    count_presentation += 1
    presentation = input()

total_average /= count_presentation
print(f"Student's final assessment is {total_average:.2f}.")
